packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx

```diff
@@ -182,40 +189,90 @@ const DialogueRecommendationRow = ({ rowProps, classes, showSuggestedTopics }: D
     skip: !currentUser
   });
 
-  if (!currentUser) return <></>;
+  const { loading: tagLoading, error: tagError, data: tagData } = useQuery(gql`
+    query UserTopTags($userId: String!) {
+      UserTopTags(userId: $userId) {
+        tag {
+          name
+          _id
+        }
+        commentCount
+      }
+    }
+  `, {
+    variables: { userId: targetUser._id },
+  });
+
+  const { loading: postsLoading, error: postsError, data: postsData } = useQuery(gql`
+    query UsersReadPostsOfTargetUser($userId: String!, $targetUserId: String!, $limit: Int) {
+      UsersReadPostsOfTargetUser(userId: $userId, targetUserId: $targetUserId, limit: $limit) {
+        _id
+        title
+      }
+    }
+  `, {
+    variables: { userId: currentUser?._id, targetUserId: targetUser._id, limit : 3 },
+  });
+
+  const topTags:[TagWithCommentCount] = tagData?.UserTopTags;
+  const readPosts:DbPost[] = postsData?.UsersReadPostsOfTargetUser
   const preTopicRecommendations: TopicRecommendationWithContents[] | undefined = topicData?.GetTwoUserTopicRecommendations; 
   const topicRecommendations = preTopicRecommendations?.filter(topic => ['agree', 'disagree'].includes(topic.theirVote) ); // todo: might want better type checking here in future for values of theirVote
-  const numRecommendations = topicRecommendations?.length ?? 0;
-  const numShown = isExpanded ? numRecommendations : 1
+ 
+  if (!currentUser || !topTags || !topicRecommendations || !readPosts) return <></>;
+  const tagsSentence = topTags.slice(0, 4).map(tag => tag.tag.name).join(', ');
+  const numRecommendations = topicRecommendations?.length + readPosts?.length + 1 ?? 0;
+  const numShown = isExpanded ? numRecommendations : 2
   const numHidden = Math.max(0, numRecommendations - numShown);
 
+  const getTopicSuggestion = (reactIconName:string, prefix:string, Content:JSX.Element) => {
+    return (
+      <div className={classes.suggestionRow}>
+        <p className={classNames({
+          [classes.debateTopicExpanded]: isExpanded,
+          [classes.debateTopicCollapsed]: !isExpanded
+        })}>
+            <ReactionIcon key={"1"} size={13} react={reactIconName} />
+            {" "}
+            <span key={"2"} className={classNames(classes.agreeText, { [classes.agreeTextCollapsedMobile]: !isExpanded})}>
```

Comment:
nit: attributes which have fixed string values (i.e. `"1"`, `"2"`) can just be written like so: `key="1"`

Reply:
@lagerros I roll to disbelieve that it saves you time to fix things like this in follow-up PRs.

Reply:
for sure, just fixed it in this one